### ISSUE TITLE
Few suggestions 

### ISSUE THREAD

**Issue by edwardyehuang** [ssyc123/TFISeg#1]
Hi, I am very interested in this work. Could you please provide the link to this paper? Suppose you need to keep the methodology secret at the moment. Please at least provide the following:

1. a figure of concept (what this work can do)
2. a few visualizations of the results
3. some experimental results on instance segmentation

**Comment by dongzhang89** [comment]
@edwardyehuang  Thank you for your interest. The paper will be published on arXiv asap.

**Comment by edwardyehuang** [comment]
Here are some quick suggestions after I read this paper:

1. Why not conduct experiments on the COCO dataset, the most popular dataset nowadays in image segmentation?
2. It seems this method can be easily expanded to panoptic segmentation.
3. How well does this method handle overlay objects (e.g., two zebras stand overlay)? Any visualizations?

**Comment by dongzhang89** [comment]
@edwardyehuang
1. We conducted experiments on two randomly selected datasets. The experiment on the COCO dataset is currently being conducted in another work. If you are interested in this work, we would appreciate it if you could report the experimental results of the COCO dataset on our method. 
2. Yes, expanding to PS should be a very intuitive idea. 
3. Please refer to the example of "person" in the third row of Figure 3 in the main paper.
4. Since we used pre-trained models from others and the backbones were first pre-trained on ImageNet before being trained on specific segmentation datasets, we wrote it this way to avoid ambiguity.
Thank you for your interest.

**Comment by edwardyehuang** [comment]
> @edwardyehuang
> 
> 1. We conducted experiments on two randomly selected datasets. The experiment on the COCO dataset is currently being conducted in another work. If you are interested in this work, we would appreciate it if you could report the experimental results of the COCO dataset on our method.
> 2. Yes, expanding to PS should be a very intuitive idea.
> 3. Please refer to the example of "person" in the third row of Figure 3 in the main paper.
> 4. Since we used pre-trained models from others and the backbones were first pre-trained on ImageNet before being trained on specific segmentation datasets, we wrote it this way to avoid ambiguity.
>    Thank you for your interest.

Thanks for your reply.

However, for question 4, I still cannot understand.

So, how many ResNet (backbone) are in your proposed model?

Currently, from Fig4. (a), I think there are two ResNet, the first one is for semantic segmentation, and the second one uses the result of semantic segmentation as the input and performs instance segmentation.

Please correct me if I am wrong.

Thanks :)

**Comment by edwardyehuang** [comment]
Or maybe, one of my other understandings, as shown below:

![image](https://github.com/ssyc123/TFISeg/assets/3354448/0cacfd86-dfcb-4a32-bce0-c0e71a20ef10)

**Comment by ssyc123** [comment]
@edwardyehuang 
I'm sorry for not understanding the arrow you drew on the figure. 

As described in our paper, TFISeg consists of two parts: semantic segmentation and instance predictions. We use trained off-the-shelf semantic segmentation without changing its network structure. That is, the backbone in semantic segmentation is not determined by us, and it is not necessarily ResNet50, for example, OCRNet in Table Ⅱ uses HRNetV2-W18/W48 as the backbone. In addition, for instance predictions, we use ResNet50 as the backbone after comprehensive consideration, as described in Sec. IV-C.

I hope my reply will be helpful.

**Comment by edwardyehuang** [comment]
> @edwardyehuang I'm sorry for not understanding the arrow you drew on the figure.
> 
> As described in our paper, TFISeg consists of two parts: semantic segmentation and instance predictions. We use trained off-the-shelf semantic segmentation without changing its network structure. That is, the backbone in semantic segmentation is not determined by us, and it is not necessarily ResNet50, for example, OCRNet in Table Ⅱ uses HRNetV2-W18/W48 as the backbone. In addition, for instance predictions, we use ResNet50 as the backbone after comprehensive consideration, as described in Sec. IV-C.
> 
> I hope my reply will be helpful.

Hi, thank you for your reply :)

Let me use another way to ask:

What is the input of the backbone of instance segmentation?

**Comment by ssyc123** [comment]
@edwardyehuang 

The semantic mask.

**Comment by edwardyehuang** [comment]
> @edwardyehuang
> 
> The semantic mask.

Thanks for the reply :)

That is what I want to ask in my original question 4: 

The semantic mask is shaped [batch, H, W, 1]. However, an imagenet-pretrained backbone (e.g., ResNet, ViT, Swin) only accepts an input with  [batch, H, W, 1].

So how can the semantic mask be passed to an imagenet-pretrained backbone for instance segmentation?

**Comment by dongzhang89** [comment]
@ssyc123  I suggest that you could release a portion of the code to help eliminate misunderstandings about this work from others. Thanks.

**Comment by ssyc123** [comment]
@edwardyehuang 
When a single-channel mask is fed into the network, we first copy it three times to convert it into a three-channel one. 

To eliminate other misunderstandings about this work, I took Dr. Zhang's advice and tried to negotiate with the other co-authors to release the code. 

Thank you for your interest.
